bundles/org.eclipse.cdt.lsp.editor.ui/src/org/eclipse/cdt/lsp/editor/ui/clangd/CProjectChangeMonitor.java

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Bachmann electronic GmbH and others.
+ * 
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ * Gesa Hentschke (Bachmann electronic GmbH) - initial implementation
+ *******************************************************************************/
+
+package org.eclipse.cdt.lsp.editor.ui.clangd;
+
+import java.io.IOException;
+
+import org.eclipse.cdt.core.CCorePlugin;
+import org.eclipse.cdt.core.settings.model.CProjectDescriptionEvent;
+import org.eclipse.cdt.core.settings.model.ICConfigurationDescription;
+import org.eclipse.cdt.core.settings.model.ICProjectDescription;
+import org.eclipse.cdt.core.settings.model.ICProjectDescriptionListener;
+import org.eclipse.cdt.lsp.editor.ui.LspEditorUiPlugin;
+import org.eclipse.cdt.lsp.editor.ui.preference.LspEditorPreferencesTester;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.yaml.snakeyaml.scanner.ScannerException;
+
+public class CProjectChangeMonitor {
+	
+	private final ICProjectDescriptionListener listener = new ICProjectDescriptionListener() {
+
+		@Override
+		public void handleEvent(CProjectDescriptionEvent event) {
+			ICProjectDescription newCProjectDecription = event.getNewCProjectDescription();
+			if (newCProjectDecription != null) {
+				IProject project = event.getProject();
+				if (project != null && LspEditorPreferencesTester.preferLspEditor(project)) {
+					ICConfigurationDescription newConfig = newCProjectDecription.getDefaultSettingConfiguration();
+					var cwdBuilder = newConfig.getBuildSetting().getBuilderCWD();
```

Comment:
Its too bad we can't just call `org.eclipse.cdt.managedbuilder.makegen.IManagedBuilderMakefileGenerator.getBuildWorkingDir()` directly, having to do the variable resolve and then split below is an unfortunate complication.